src/components/ui/Tag/index.js

```diff
@@ -0,0 +1,147 @@
+
+  return (
+    <span ref={ref} {...otherProps}>
+      {label && (
+        <span // TODO: switch with VisuallyHidden component, once available.
+          css={css`
+            font-style: bold;
+            position: absolute;
+            left: -10000px;
+            top: auto;
+            width: 1px;
+            height: 1px;
+            overflow: hidden;
+          `}
+        >
+          {label}:
+        </span>
+      )}
+
+      {children}
+      {onDismiss && (
+        <button
+          type="button"
+          onClick={onDismiss}
+          css={css`
+            box-shadow: 0;
+            border: 0;
+            background: none;
+            color: currentColor;
+            cursor: pointer;
+            vertical-align: 0.2rem;
+            margin: 0;
+            padding: 0;
+
+            &:hover {
+              opacity: 0.75;
+            }
+          `}
+        >
+          <Icon
+            name="x"
+            small
+            css={css`
+              display: inline-block;
+              margin-left: ${toUnits(theme.spacing.padding.small)};
+              margin-top: ${toUnits(theme.spacing.padding.xxSmall)};
+              opacity: 1;
+              vertical-align: middle;
+            `}
+          />
+        </button>
+      )}
+    </span>
+  );
+});
+
+export const styles = (props) => {
+  const { color, count, status, theme } = props;
+  let textColor = theme.colors.text.light;
+  let backgroundColor;
+
+  if (color) {
+    backgroundColor = color;
+  }
+
+  if (status) {
+    backgroundColor = theme.colors.intent[status];
+  }
+
+  if (status || color) {
+    textColor = getContrastingTextColor({ color: backgroundColor, theme });
+  }
+
+  return css`
+    ${metadata.small(theme)};
+    color: ${textColor};
+    margin: 0 ${toUnits(theme.spacing.padding.xSmall)}
+      ${toUnits(theme.spacing.padding.xSmall)} 0;
+
+    ${backgroundColor &&
+      css`
+        background: ${backgroundColor};
+        padding: ${toUnits(theme.spacing.padding.xSmall)}
+          ${toUnits(theme.spacing.padding.small)};
+      `}
+
+    ${count &&
+      css`
+        border-radius: ${toUnits(theme.spacing.padding.xLarge)};
+        min-width: ${toUnits(theme.spacing.padding.xLarge)};
+        height: ${toUnits(theme.spacing.padding.xLarge)};
+        display: inline-flex;
+        align-items: center;
+        justify-content: center;
+        box-sizing: border-box;
+      `}
+  `;
+};
+
+Tag.defaultProps = {
+  children: undefined,
+  color: null,
+  label: null,
+  count: false,
+  onDismiss: null,
+  status: undefined,
+};
+
+Tag.propTypes = {
+  /** @ignore */
+  children: PropTypes.node,
+  /** Use colour to differentiate different tags or indicate status. */
+  color: PropTypes.string,
+  /** Optional label for providing additional context to screen reader users. */
+  label: PropTypes.string,
+  /** Style numbers in a more badge-y kind of way. */
+  count: PropTypes.bool,
```

Comment:
Count is the wrong prop name I think. Why not `badge`?

Reply:
I suppose we could do that. My concern is that this then starts bleeding into the styling, whereas the intention here is to use it for counts or tallies or numbers. I'm trying to aim for more semantic, rather than stylistic, names, but badge is commonly enough used by other systems that I'm okay with going more relaxed about it for now.

Reply:
But it's a boolean. If this is supposed to take a number wouldn't the _content_ be a number? This is just about styling from what I understand.

Reply:
No no, it's not supposed to take a number—it's just a property of "this is a count". (It was originally called `numerical` but I changed it out for being too wordy.) The idea here is that "badge" only refers to how the styling is applied, but "count" or "number" refers to actual semantic usage—like "sidebar" rather than "blue-box" in CSS classes, for instance.

I could see how that might be confusing though. Alternative suggestions I'm fine with using here:
- badge
- number
- numeric
- tally

Reply:
Yeah, count and tally are definitely misleading and I find them confusing. I think `badge` is the right approach. What semantic value is there in saying the kind of content here? And why only offer this one option? I think it's rooted in the right place but `badge` is the right approach here.

`isNumeric` or `isNumber` could work otherwise but I don't love that. I'd say stick with `badge`. 🤷‍♂